portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,8 +49,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       long id = entity.getKey().getId();
       String message = (String) entity.getProperty("message");
       long timestamp = (long) entity.getProperty("timestamp");
-
-      Message m = new Message(id, message, timestamp);
+      double score;
+      if (entity.getProperty("score") == null) {
+        score = -2;
```

Comment:
first note: this works, so I'm not going to hold up this PR for it, but something to think about...

it's generally best to avoid "magic" values like this, particularly for error handling. Especially since the library already uses a more commonly understood magic value (`null`) for this.

Could we use `null` throughout, here? It would definitely look more natural in the JS code to be comparing against `null` instead of 2.

Of course, this depends on whether or not Gson can properly serialize nulls into JSON. I didn't know it's behavior, so I Googled for "gson serialize null" and found some potentially interesting results.

If you choose to keep this magic value, I'd suggest two things to make this more understandable:

- store it in a class-level constant, e.g.,  `private static final INVALID_SENTIMENT_SCORE = 2;`
- add some comments to describe why this case is having to be handled.

Reply:
I couldn't figure out how to use `null` properly. When I tried to do so, it would result in an error. However, I created a class-level constant with comments to make it look cleaner.